packages/ember-application/tests/system/visit_test.js

```diff
@@ -142,7 +142,9 @@ moduleFor('Ember.Application - visit()', class extends ApplicationTestCase {
        * Visit on the application a second time. The application should remain
        * booted, but a new instance will be created.
        */
-      return this.visit('/');
+      return this.application.visit('/').then(instance => {
+        this.applicationInstance = instance;
```

Comment:
isn't this already set?

Reply:
No, just above we destroyed it and set it to null in this specific test.  This test is intentionally confirming that calling `this.application.visit()` will create _another_ application instance...